crates/astria-sequencer/src/app/mod.rs

```diff
@@ -758,18 +791,24 @@ impl App {
         Ok(())
     }
 
-    /// Executes the given block, but does not write it to disk.
-    ///
-    /// `commit` must be called after this to write the block to disk.
+    /// updates the app state after transaction execution, and generates the resulting
+    /// `SequencerBlock`.
     ///
-    /// This is called by cometbft after the block has already been
-    /// committed by the network's consensus.
-    #[instrument(name = "App::finalize_block", skip_all)]
-    pub(crate) async fn finalize_block(
+    /// this must be called after a block's transactions are executed.
+    #[instrument(name = "App::post_execute_transactions", skip_all)]
+    async fn post_execute_transactions(
         &mut self,
```

Comment:
Hard to follow all the side effects in this function.

Reply:
this function is basically everything that was in `finalize_block` after txs were executed (minus mempool cleaning, as that needs to be done only when the block is actually finalized)